src/Resources/contao/classes/DataContainer.php

```diff
@@ -902,8 +902,9 @@ protected function initPicker()
 
 		$this->setPickerValue();
 
-		$strDriver = 'DC_' . $GLOBALS['TL_DCA'][$this->strPickerTable]['config']['dataContainer'];
-		$objDca = new $strDriver($this->strPickerTable);
+		// Create a DataContainer object (see #886)
+		$objDca = new \DataContainer();
```

Comment:
You are wrong. The callbacks are executed for the real DCA object, making sure that all permissions are checked and that the user only sees their allowed modules. The helper object is only used in the callbacks.

> because most of the time the `strPickerTable` will be different for the popup

That's exactly why we **must not** trigger the callbacks. You cannot run the permission checks for `target=tl_content.singleSRC.2` in e.g. an `do=news` environment.